src/serializer/nodeSerializers.ts

```
@@ -0,0 +1,402 @@
+        return '@'
+      case 'Parent':
+        return '^'.repeat(node.n)
+      case 'Parameter':
+        return `$${node.name}`
+      case 'AccessAttribute':
+        return this.serializeAccessAttribute(node)
+      case 'AccessElement':
+        return this.serializeAccessElement(node)
+      case 'Array':
+        return this.serializeArray(node)
+      case 'ArrayCoerce':
+        return this.serializeArrayCoerce(node)
+      case 'Object':
+        return this.serializeObject(node)
+      case 'OpCall':
+        return this.serializeOpCall(node)
+      case 'And':
+        return this.serializeBinaryOp(node.left, '&&', node.right)
+      case 'Or':
+        return this.serializeBinaryOp(node.left, '||', node.right)
+      case 'Not':
+        return this.serializeUnaryOp('!', node.base)
+      case 'Neg':
+        return this.serializeUnaryOp('-', node.base)
+      case 'Pos':
+        return this.serializeUnaryOp('+', node.base)
+      case 'Group':
+        return this.serializeGroup(node)
+      case 'FuncCall':
+        return this.serializeFuncCall(node)
+      case 'PipeFuncCall':
+        return this.serializePipeFuncCall(node)
+      case 'Deref':
+        return this.serializeDeref(node)
+      case 'Filter':
+        return this.serializeFilter(node)
+      case 'Projection':
+        return this.serializeProjection(node)
+      case 'Slice':
+        return this.serializeSlice(node)
+      case 'InRange':
+        return this.serializeInRange(node)
+      case 'Select':
+        return this.serializeSelect(node)
+      case 'Asc':
+        return this.serializeAsc(node)
+      case 'Desc':
+        return this.serializeDesc(node)
+      case 'Tuple':
+        return this.serializeTuple(node)
+      case 'Map':
+        return this.serializeMap(node)
+      case 'FlatMap':
+        return this.serializeFlatMap(node)
+      case 'Context':
+        return this.serializeContext(node)
+      case 'Selector':
+        return '<selector>'
```

better to throw an error than to generate invalid groq:
```suggestion
        throw new Error('Can't serialize Selector')
```